src/main/java/com/monumental/models/Monument.java

```diff
@@ -258,4 +267,108 @@ private String generateDescription() {
 
         return description;
     }
+
+    /**
+     * Adds a Contribution to the List
+     * Will make a new ArrayList if this.contributions is null
+     * Checks if a Contribution has already been added to this.contributions with the same date and name
+     * Will do nothing if the specified Contribution has a null date or null/empty submittedBy
+     * and does nothing if so
+     * @param contribution - Contribution to add to the List
+     */
+    public void addContribution(Contribution contribution) {
+        if (this.contributions == null) {
+            this.contributions = new ArrayList<>();
+        }
+
+        if (contribution.getDate() == null || contribution.getSubmittedBy() == null ||
+                contribution.getSubmittedBy().isEmpty()) {
+            return;
+        }
+
+        for (Contribution c : this.contributions) {
+            if (c.getDate() == contribution.getDate() &&
+                    c.getSubmittedBy().equalsIgnoreCase(contribution.getSubmittedBy())) {
+                return;
+            }
+        }
+
+        this.contributions.add(contribution);
+    }
+
+    /**
+     * Adds a Reference to the List
+     * Will make a new ArrayList if this.references is null
+     * Will do nothing if the specified Reference has a null/empty url
+     * Checks if a Reference has already been added to this.references with the same url
+     * and does nothing if so
+     * @param reference - Reference to add to the List
+     */
+    public void addReference(Reference reference) {
+        if (this.references == null) {
+            this.references = new ArrayList<>();
+        }
+
+        if (reference.getUrl() == null || reference.getUrl().isEmpty()) {
+            return;
+        }
+
+        for (Reference r : this.references) {
+            if (r.getUrl().equalsIgnoreCase(reference.getUrl())) {
+                return;
+            }
+        }
+
+        this.references.add(reference);
+    }
+
+    /**
+     * Encapsulates the logic to validate a Monument object
+     * Use this method to manually run validation in lieu of a @Valid Spring annotation
+     * @return ValidationResult - ValidationResult object representing the result of the validation
+     */
+    public ValidationResult validate() {
```

Comment:
Why is this needed?

Reply:
It's used in the `InitialDataLoadApplication`, but they're probably not needed right? The validation should run on insert/update?

Reply:
I'm not sure. They run on the controller methods so maybe not. But maybe there is a way to use `@Valid` outside of controller method params.